onnxruntime/core/optimizer/propagate_cast_ops.cc

```diff
@@ -1463,13 +1497,13 @@ Status PropagateCastOps::ApplyImpl(Graph& graph, bool& modified, int graph_level
 }
 
 PropagateCastOps::PropagateCastOps(GraphTransformerConfiguration::PropagateCastOpsConfiguration::Strategy strategy,
-                                   size_t level, const std::vector<std::string>& allow_list,
-                                   const std::unordered_set<std::string>& compatible_execution_providers) noexcept
-    : GraphTransformer("PropagateCastOps", compatible_execution_providers), level_(level),
-      fp16_allow_ops_{std::unordered_set<std::string>(allow_list.begin(), allow_list.end()),
-		      std::unordered_set<std::string>(level1_fp16_allow_ops.begin(), level1_fp16_allow_ops.end()),
-		      std::unordered_set<std::string>(level2_fp16_allow_ops.begin(), level2_fp16_allow_ops.end())},
-      strategy_(strategy) {
+                                   size_t level, gsl::span<const std::string> allow_list,
+                                   const InlinedHashSet<std::string_view>& compatible_execution_providers)
+    : GraphTransformer("PropagateCastOps", compatible_execution_providers), level_(level), fp16_allow_ops_0_(), strategy_(strategy) {
+  if (!allow_list.empty()) {
```

Comment:
could allow_list just be used to initialize fp16_allow_ops_0_ directly in the member initializer list?